### PR TITLE
Support automatic coda dataset creation

### PR DESCRIPTION
**Pull request by DanielMwendwa**

### CODE REVIEW

File: coda_v2_python_client/firebase_client_wrapper.py [hunk 1]
```diff
@@ -699,6 +706,26 @@ def set_dataset_user_ids(self, dataset_id, user_ids):
...
+    def update_dataset_user_ids(self, dataset_id, user_ids):
+        """
+        Updates user ids for the given dataset.
+
+        :param dataset_id: Id of a dataset.
+        :type dataset_id: str
+        :param user_ids: list of user ids.
+        :type user_ids: list
+        """
+        if len(user_ids) > 0:
```
Comment (as2388) [inline coda_v2_python_client/firebase_client_wrapper.py:718]: Can you log a warning if this isn't true?
e.g. 
```
if len(user_ids) == 0:
    log.warning(...)
    return
```

File: coda_v2_python_client/firebase_client_wrapper.py [hunk 2]
```diff
@@ -678,7 +678,14 @@ def get_dataset_user_ids(self, dataset_id, transaction=None):
...
         self.ensure_user_ids_consistent(dataset_id, transaction=transaction)
-        return self.get_segment(dataset_id, transaction=transaction).get("users")
+
+        segment_snapshot = self.get_segment(dataset_id, transaction=transaction)
+        segment_doc = segment_snapshot.to_dict()
+
+        user_ids = []
+        if segment_doc is not None and "users" in segment_doc:
+            user_ids = segment_snapshot.get("users")
+        return user_ids
```
Comment (as2388) [inline coda_v2_python_client/firebase_client_wrapper.py:688]: If the document or users don't exist should we return None or []?

**Review by as2388 — APPROVED** [review]

File: coda_v2_python_client/firebase_client_wrapper.py (hunk 2 above)
Comment (DanielMwendwa) [reply coda_v2_python_client/firebase_client_wrapper.py]: updated the function to return None.

File: coda_v2_python_client/firebase_client_wrapper.py (hunk 1 above)
Comment (DanielMwendwa) [reply coda_v2_python_client/firebase_client_wrapper.py]: updated, thanks.

**Pull request merged**